unittest/force-styles/check_tests.py

```diff
@@ -0,0 +1,224 @@
+                counter += 1
+    return counter
+
+print("Parsing style names from C++ tree in: ",src)
+
+for h in headers:
+    if verbose: print("Checking ", h)
+    fp = open(h)
+    text = fp.read()
+    fp.close()
+    matches = re.findall("(.+)Style\((.+),(.+)\)",text,re.MULTILINE)
+    for m in matches:
+
+        # skip over internal styles w/o explicit documentation
+        style = m[1]
+        total += 1
+        if upper.match(style):
+            continue
+
+        # detect, process, and flag suffix styles:
+        info = { 'kokkos':  0, 'gpu':     0, 'intel':   0, \
+                 'omp':     0, 'opt':     0, 'removed': 0 }
+        suffix = kokkos_skip.match(style)
+        if suffix:
+            continue
+        suffix = gpu.match(style)
+        if suffix:
+            style = suffix.groups()[0]
+            info['gpu'] = 1
+        suffix = intel.match(style)
+        if suffix:
+            style = suffix.groups()[0]
+            info['intel'] = 1
+        suffix = kokkos.match(style)
+        if suffix:
+            style = suffix.groups()[0]
+            info['kokkos'] = 1
+        suffix = omp.match(style)
+        if suffix:
+            style = suffix.groups()[0]
+            info['omp'] = 1
+        suffix = opt.match(style)
+        if suffix:
+            style = suffix.groups()[0]
+            info['opt'] = 1
+        deprecated = removed.match(m[2])
+        if deprecated:
+            info['removed'] = 1
+
+        # register style and suffix flags
+        if m[0] == 'Angle':
+            register_style(angle,style,info)
+        elif m[0] == 'Bond':
+            register_style(bond,style,info)
+        elif m[0] == 'Dihedral':
+            register_style(dihedral,style,info)
+        elif m[0] == 'Improper':
+            register_style(improper,style,info)
+        elif m[0] == 'KSpace':
+            register_style(kspace,style,info)
+        elif m[0] == 'Pair':
+            register_style(pair,style,info)
+
+
+counter = 0
+
+def check_tests(name,list,yaml,search,skip=()):
+    num = 0
+    yaml_files = glob(os.path.join(tests, yaml))
+    styles = []
+    missing = []
+    for y in yaml_files:
+        if verbose: print("Checking: ",y)
+        fp = open(y)
+        text = fp.read()
+        fp.close()
+        matches = re.findall(search,text,re.MULTILINE)
+        for m in matches:
+            if m[1] == 'hybrid' or m[1] == 'hybrid/overlay':
+                for s in m[0].split():
+                    if s in list.keys():
+                        styles.append(s)
+            else:
+                styles.append(m[1])
+    for s in list.keys():
+        # known undocumented aliases we need to skip
+        if s in skip: continue
+        if not s in styles:
+            if not list[s]['removed']:
+                if verbose: print("No test for %s style %s" % (name,s))
+                num += 1
+                missing.append(s)
+    total = len(list)
```

Comment:
The list of missing tests would be more readable if it's sorted alphabetically

Reply:
done